azurerm/internal/services/bot/bot_channel_skype_resource.go

```diff
@@ -0,0 +1,232 @@
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.BotChannelID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.BotServiceName, string(botservice.ChannelNameSkypeChannel))
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] %s was not found - removing from state!", id)
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("retrieving %s: %+v", id, err)
+	}
+
+	d.Set("bot_name", id.BotServiceName)
+	d.Set("resource_group_name", id.ResourceGroup)
+	d.Set("location", location.NormalizeNilable(resp.Location))
+
+	if props := resp.Properties; props != nil {
+		if channel, ok := props.AsSkypeChannel(); ok {
+			if channelProps := channel.Properties; channelProps != nil {
+				d.Set("calling_web_hook", channelProps.CallingWebHook)
+				d.Set("enable_calling", channelProps.EnableCalling)
+				d.Set("enable_groups", channelProps.EnableGroups)
+				d.Set("enable_media_cards", channelProps.EnableMediaCards)
+				d.Set("enable_messaging", channelProps.EnableMessaging)
+			}
+		}
+	}
+
+	return nil
+}
+
+func resourceBotChannelSkypeUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Bot.ChannelClient
+	ctx, cancel := timeouts.ForUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.BotChannelID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	parameters := botservice.BotChannel{
+		Properties: botservice.SkypeChannel{
+			Properties: &botservice.SkypeChannelProperties{
+				EnableCalling:    utils.Bool(d.Get("enable_calling").(bool)),
+				EnableGroups:     utils.Bool(d.Get("enable_groups").(bool)),
+				EnableMediaCards: utils.Bool(d.Get("enable_media_cards").(bool)),
+				EnableMessaging:  utils.Bool(d.Get("enable_messaging").(bool)),
+				IsEnabled:        utils.Bool(true),
+			},
+			ChannelName: botservice.ChannelNameBasicChannelChannelNameSkypeChannel,
+		},
+		Location: utils.String(azure.NormalizeLocation(d.Get("location").(string))),
+		Kind:     botservice.KindBot,
+	}
+
+	if v, ok := d.GetOk("calling_web_hook"); ok {
+		channel, _ := parameters.Properties.AsSkypeChannel()
+		channel.Properties.CallingWebHook = utils.String(v.(string))
+	}
+
+	if _, err := client.Update(ctx, id.ResourceGroup, id.BotServiceName, botservice.ChannelNameSkypeChannel, parameters); err != nil {
+		return fmt.Errorf("updating %s: %+v", id, err)
+	}
+
+	return resourceBotChannelSkypeRead(d, meta)
+}
+
+func resourceBotChannelSkypeDelete(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Bot.ChannelClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.BotChannelID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Delete(ctx, id.ResourceGroup, id.BotServiceName, string(botservice.ChannelNameSkypeChannel))
```

Comment:
If the service creates a default, should we restore this here, rather than deleting?

Reply:
When the `callingWebhook` property is never used, I can directly restore it using update operation. But when this property is specified, I have to recreate it for restoring it with default settings. Because there is a [bug](https://github.com/Azure/azure-rest-api-specs/issues/15170) on the `callingWebhook` property, which means it cannot be updated from a valid value to empty value once it is changed to a valid value "xxxx".